ignite/services/plugin/client_api.go

```diff
@@ -0,0 +1,41 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/ignite/cli/ignite/pkg/cosmosanalysis/chain"
+	chainservice "github.com/ignite/cli/ignite/services/chain"
+)
+
+// NewClientAPI creates a new app ClientAPI.
+func NewClientAPI(c *chainservice.Chain) ClientAPI {
+	return clientAPI{chain: c}
+}
+
+type clientAPI struct {
+	chain *chainservice.Chain
+}
```

Comment:
We should consider using an interface here because `Chain` has a lot of methods, many of which we won't be using:

```suggestion
import (
	"context"
	"encoding/json"

	chainconfig "github.com/ignite/cli/ignite/config/chain"
	"github.com/ignite/cli/ignite/pkg/cosmosanalysis/chain"
)

type Chainer interface {
	// AppPath returns the configured App's path.
	AppPath() string

	// Config returns the config of the chain.
	Config() (*chainconfig.Config, error)
}

// NewClientAPI creates a new app ClientAPI.
func NewClientAPI(c Chainer) ClientAPI {
	return clientAPI{chain: c}
}

type clientAPI struct {
	chain Chainer
}
```

That will also help with testing/mocking.